### DISCUSSION TITLE
How to get response data on mutation error ?

### DISCUSSION THREAD

**Discussion by aswinmohanme** [TanStack/query#1846] [Q&A] [answered]
I have a sign in mutation, that returns a 422 response when the credentials are not correct, along with a data object that contains the error message for the keys.

```
function signInWithPassword() {
  return useMutation("sign_in_mutation", ({ email, password }) =>
    axiosApi.post("/login", {
      email: email,
      password: password,
      device_token: "device_token",
      device_uuid: "device_uuid",
    }),
  )
}
```

Here is the part where the mutation is called.
```
              signInMutation.mutate(
                { email: userStore.email, password: passwordText },
                {
                  onSuccess: async (data, _variables, _context) => {
                    const token = data.data.token
                    if (token) {
                      await saveToken(token)
                      setAuthToken(token)
                      userStore.setLoggedIn(true)
                    }
                  },
                },
              )

```

When the mutation errors out due to incorrect credentials, the onError property of the mutation gets set to true. But the error property is undefined. 

I read on the documentation that there is no need for custom errors throws on Api Clients like axios. Why is the error property undefined and how can I set it up to get the error response send by the Server. For reference here is the server response

```
{
  "message": "The given data was invalid.",
  "errors": {
    "password": [
      "These credentials do not match our records."
    ]
  }
}
```

**Comment by TkDodo** [comment, marked as answer]
Do you have a codesandbox? I’m pretty sure that works. 

> When the mutation errors out due to incorrect credentials, the onError property of the mutation gets set to true. But the error property is undefined.

not sure what you mean. onError is a callback function like onSuccess 🤔

**Reply by aswinmohanme** [reply]
Sorry I meant isError property, but the signInMutation.error property is undefined. I am using Axios with React Native.

**Reply by TkDodo** [reply]
`.isError` and `.error` should always correlate, so if you have `isError` set to true, your `error` should be defined. Hard to say more without seeing some runnable code.
